### PR TITLE
Compatibility with System Integrity Protection on MacOS

### PR DESCRIPTION
**Pull request by srherbener**
## Description

This PR changes the direct calls to `iodaconv_comp.sh` with calls to `bash iodaconv_comp.sh ...` so that the System Integrity Protection (SIP) on MacOS can remain enabled. SIP helps protect Macs from malware and viruses so it is good to keep that enabled. See jcsda-internal/ioda/issus/581 for more details about what is being addressed.

### Issue(s) addressed

- fixes jcsda-internal/ioda/issues/581

## Acceptance Criteria (Definition of Done)

ioda converter ctests all pass on a Mac with SIP enabled.

And idoa converter ctests continue passing on HPC systems. Need to test these systems:
- [x] Orion (Steve)
- [x] Cheyenne (Jaime)
- [x] S4 (Steve)
- [x] Hera

## Dependencies

None

## Impact

None

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -771,122 +810,137 @@ if(iodaconv_bufr_ENABLED)
 
   ecbuild_add_test( TARGET  test_iodaconv_bufr_hrs2ioda
                     TYPE    SCRIPT
-                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
-                    ARGS    netcdf
+                    COMMAND bash
+                    ARGS    ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                            netcdf
```
Comment (jamiebresch) [inline test/CMakeLists.txt:815]: Why is test_iodaconv_bufr_mhs2ioda (line 805) different from others?

**Review by jamiebresch — COMMENTED** [review]
After `brew install bash`, most of the previously failed iodaconv ctests passed on my Mac.

File: test/CMakeLists.txt (hunk 1 above)
Comment (srherbener) [reply test/CMakeLists.txt:815]: @jamiebresch thanks for catching this! It's a typo and I'll fix it.

Comment (srherbener) [reply test/CMakeLists.txt:816]: Done!

**Review by jamiebresch — APPROVED** [review]
Adding `bash` command has no impact on Cheyenne. iodaconv ctests are not all passing on Cheyenne, but the failed iodaconv ctests are not related to this PR.

**Comment by srherbener** [comment]
I've verified that this works on Orion and S4. I built ioda-bundle with converters enabled, and all tests pass.

@CoryMartin-NOAA I don't have access to Hera. Do you want Hera tested before merging, and if so could you or a colleague do that testing? Thanks!

**Comment by climbfuji** [comment]
I can test on hera if you tell me which tests you want me to run.

**Comment by srherbener** [comment]
> I can test on hera if you tell me which tests you want me to run.

@climbfuji, I was running: `ctest -R ioda` in my batch script which runs all ioda and ioda-converter tests. Thanks for helping with this!

**Comment by climbfuji** [comment]
This worked on hera (running on the login nodes using mpirun and being yelled at by the process scanners).

**Comment by climbfuji** [comment]
@srherbener I have not forgotten this, testing it on macOS today.

**Comment by srherbener** [comment]
@climbfuji no worries, I know there's a lot going on and I appreciate your help!

**Review by climbfuji — APPROVED** [review]
I have tested this to the bones and it does what it should. I will have similar changes for other repositories later today.

**Comment by srherbener** [comment]
@climbfuji thank you for your thorough testing and help with this! It's much appreciated.

**Pull request merged**